lib/ddtrace/contrib/rack/middlewares.rb

```diff
@@ -22,24 +23,48 @@ class TraceMiddleware
           value
         end
         option :distributed_tracing, default: false
+        option :request_queuing, default: false
+        option :web_service_name, default: 'web-server', depends_on: [:tracer, :request_queuing] do |value|
+          if get_option(:request_queuing)
+            get_option(:tracer).set_service_info(value, 'webserver', Ext::AppTypes::WEB)
+          end
+          value
+        end
 
         def initialize(app)
           @app = app
         end
 
+        def compute_queueing_time(env, tracer)
+          return unless Datadog.configuration[:rack][:request_queuing]
+
+          # parse the request enqueue time
+          request_start = Datadog::Contrib::Rack::QueueTime.get_request_start(env)
+          return if request_start.nil?
+
+          Datadog::Tracer.log.debug('[experimental] creating request.enqueuing span')
+          tracer.trace(
+            'request.enqueuing',
```

Comment:
Nit: the name doesn't align with our usual informal conventions (`request.` looks like it comes from `request` library).
I also guess it has to stay generic between web servers (not Nginx only).
So maybe something like `web_server.start_request`, `http_server.start_request`?
I also saw that there is a potential for another `X-Queue-Start` timing, it would then easily translate into a `whatever.queue_start`?

Reply:
totally agree, we can change that value with one you've suggested!

Reply:
If there's an HTTP header named `X-Queue-Start`, those usually get tagged as `http`. I'd suggest adding `http`, like `http.queue_start`, `http.headers.queue_start` or something similar (to indicate its origins)